### PR TITLE
Adds a bunch of interaction related unit tests

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

I made these a year ago and meant to PR them but never got around to it

Basically just adds a bunch of unit tests for a bunch of specific attack chain interactions that are prone to breaking due to snowflake

Increasing coverage of attack chain is good so people who are working on it don't have to worry about the 1000 edge cases breaking

- Blocking (both unarmed and armed)
- Cuffs
- Eyestab
- Flashes (combat mode, non combat mode, with flash protection)
- Help intent
- Pistol whip
- Butchering with a bayonet 
- Damp rag smothering
- Droppers
- EMP flashlights
- Holofans
- Door attack hand redirector
- Tool usage on Cyborgs
- Punching Cyborgs
- Ability to bash tables
- Ability to bash any structure
- Ability to use tools on machinery
- Kinetic Crusher projectile
- Spraycans (capping, making graffiti) 
- Loading a syringe gun

### CODE REVIEW

**Comment by MrMelbert** [comment]
For anyone curious these interactions were the ones that did not innately work when testing Swing

**Comment by SyncIt21** [comment]
Not related to this PR, how do you tick unit test file? By default it adds it to tgstation.dme  and not _unit_tests.dm

**Comment by MrMelbert** [comment]
I just do it manually, I don't know if we have an extension to auto do it

**Comment by SyncIt21** [comment]
> I just do it manually, I don't know if we have an extension to auto do it

Pain. We have linters to check if they are alphabetically sorted but not tools to help us sort them.

I'll try to make a python script that can be run locally for this or ask sans for help.

File: code/modules/unit_tests/_unit_tests.dm [hunk 1]
```diff
@@ -83,6 +83,8 @@
 #endif
 /// A trait source when adding traits through unit tests
 #define TRAIT_SOURCE_UNIT_TESTS "unit_tests"
+/// Helper to allocate a new object with the implied type (the type of the variable it's assigned to) in the corner of the test room
+#define EASY_ALLOCATE(arguments...) allocate(__IMPLIED_TYPE__, run_loc_floor_bottom_left, ##arguments)
```
Comment (san7890) [inline code/modules/unit_tests/_unit_tests.dm:87]: this is cool thank you

**Pull request merged**